integritybackend/actions.py

```diff
@@ -228,68 +260,131 @@ def archive(self, zip_path: str, org_config: dict, collection_id: str):
+                        else:
+                            _logger.error("Content registration on ISCN failed")
+                else:
+                    _logger.info("Content registration on ISCN skipped")
+            except Exception as e:
+                _logger.error(f"Content registration on ISCN failed: {e}")
+
+            # Register encrypted ZIP on Numbers Protocol
+            numbers_receipt = None
+            try:
+                if action_params["registration_policies"]["numbersprotocol"]["active"]:
+                    with open(extracted_meta_content) as meta_content_f:
+                        meta_content = json.load(meta_content_f)["contentMetadata"]
+                        asset_extras = {
+                            "author": meta_content["author"],
+                            "usageInfo": "Encrypted with AES-256.",
+                            "keywords": [org_id, collection_id],
+                            "extras": meta_content["extras"],
+                            "contentFingerprints":[
+                                f"hash://sha256/{enc_zip_sha}",
+                                f"hash://md5/{enc_zip_md5}",
+                                f"ipfs://{enc_zip_cid}",
+                            ],
+                            "relatedContent":[
+                                {
+                                    "value": f"hash://sha256/{content_sha}",
+                                    "description": "The SHA-256 of the original content.",
+                                },
+                                {
+                                    "value": f"hash://md5/{content_md5}",
+                                    "description": "The MD5 of the original content.",
+                                },
+                                {
+                                    "value": f"ipfs://{content_cid}",
+                                    "description": "The CID of the original content.",
+                                },
+                                {
+                                    "value": f"hash://sha256/{zip_sha}",
+                                    "description": "The SHA-256 of the unencrypted archive.",
+                                },
+                                {
+                                    "value": f"hash://md5/{zip_md5}",
+                                    "description": "The MD5 of the unencrypted archive.",
+                                },
+                                {
+                                    "value": f"ipfs://{zip_cid}",
+                                    "description": "The CID of the unencrypted archive.",
+                                },
+                            ]
+                        }
+                        numbers_receipt = Numbers.register(
+                                meta_content["name"],
+                                meta_content["description"],
+                                enc_zip_cid,
```

Comment:
All I have to do is hardcode this to the "working CID" (`bafybeif3ctgbmiso4oykvwj6jebyrkjxqr26bfrkesla5yr2ypgx47wgle`) and this succeeds:

```
[2022-06-04T07:09:12-0400] {numbers.py:80} INFO: Numbers registration succeeded: {
    "status": "success",
    "response": {
        "txHash": "0x76c77845f32d2438a45cff347b72e8308f4204ff47e84e1ccac8d844305cd902",
        "assetCid": "bafybeif3ctgbmiso4oykvwj6jebyrkjxqr26bfrkesla5yr2ypgx47wgle",
        "assetTreeCid": "bafkreie6cj6h3fx5qtfdislpvau5lqev4prh2wirbznoi4r6zlw3pzzlkq"
    }
}
```

Let me know what's going on pls.